docs/architecture/glossary.rst

```diff
@@ -7,66 +7,129 @@ Glossary
 
-  configurable logic block
   CLB
-    Basic building block of logic. 
-
-  frame
-    Fundamental unit of configuration data consisting of 101 :term:`words <word>`.
-
-  half
+  Configurable logic block (CLB)
+    The configurable logic unit of an FPGA. Also called a **logic cell**.
+    A CLB is a combination of basic logic elements (:term:`BELs <BEL>`).
+
+  Database
+    Text files containing meaningful labels for bit positions within
+    :term:`segments <segment>`.
+
+  Frame
+    The fundamental unit of :term:`bitstream` configuration data consisting of
+    101 :term:`words <word>`.
+    Each frame has a 32-bit frame address and 101 payload words, 32 bits each.
+    The 50th payload word is an EEC.
+    The 7 LSB bits of the frame address are the frame index within the
+    configuration :term:`column` (called *minor frame address* in the Xilinx
+    documentation). The rest of the frame address identifies the configuration
+    column (called *base frame address* in Project X-Ray nomenclature).
+
+    The bits in an individual frame are spread out over the entire column.
+    For example, in a logic column with 50 tiles, the first tile is configured
+    with the first two words in each frame, the next tile with the next two
+    words, and so on.
+    
+  Frame base address
+    The first configuration frame address for a :term:`column`. A frame base
+    address has always the 7 LSB bits cleared.
+
+  Fuzzer
+    Scripts and a makefile to generate one or more :term:`specimens <specimen>`
+    and then convert the data from those specimens into a :term:`database`.
+
+  Half
     Portion of a device defined by a virtual line dividing the two sets of global
-    clock buffers present in a device. The two halves are simply referred to as
+    clock buffers present in a device. The two halves are referred to as
     the top and bottom halves.
 
-  node
-    Collection of :term:`wires <wire>` spanning one or more tiles.
+  Node
+    A routing node on the device. A node is a collection of :term:`wires <wire>`
+    spanning one or more :term:`tiles <tile>`.
+    Nodes that are local to a tile map 1:1 to a wire. A node that spans multiple
+    tiles maps to multiple wires, one in each tile it spans.
 
-  programmable interconnect point
   PIP
+  Programmable interconnect point (PIP)
     Connection point between two wires in a tile that may be enabled or
     disabled by the configuration.
 
-  horizontal clock row
+  Horizontal clock row
     Portion of a device including 12 horizontal clocks and the 50 interconnect
     and function tiles associated with them. A :term:`half` contains one or
     more horizontal clock rows and each half may have a different number of
     rows.
-
-  site
-    Portion of a tile where :term:`BELs <BEL>` can be placed.  :term:`Slices
-    <slice>` in a :term:`CLB` tile are sites.
-
-  slice
-    Portion of a :term:`CLB` tile that contains :term:`BELs <BEL>`.
-
-  tile
+    
+  ROI
+  Region of interest (ROI)
```

Comment:
Same as above.

Reply:
Thanks! I've updated this one, and also the PIP entry above.